packages/wrangler/templates/startDevWorker/ProxyWorker.ts

```diff
@@ -0,0 +1,250 @@
+import assert from "node:assert";
+import type {
+	ProxyWorkerIncomingRequestBody,
+	ProxyWorkerOutgoingRequestBody,
+	ProxyData,
+} from "../../src/api/startDevWorker/events";
+import {
+	DeferredPromise,
+	createDeferredPromise,
+} from "../../src/api/startDevWorker/utils";
+
+interface Env {
+	PROXY_CONTROLLER: Fetcher;
+	PROXY_CONTROLLER_AUTH_SECRET: string;
+	DURABLE_OBJECT: DurableObjectNamespace;
+}
+
+// request.cf.hostMetadata is verbose to type using the workers-types Request -- this allows us to have Request correctly typed in this scope
```

Comment:
Isn't it just `Request<ProxyWorkerIncomingRequestBody>`? If the ergonomics of workers-types is such that it's unusable, that's probably something we should fix

Reply:
That doesn't work. It is something we should fix. I think @mrbbot said a fix was incoming?